src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs

```diff
@@ -71,9 +75,32 @@ public override void Trace(TraceEvent traceEvent)
             if (string.IsNullOrEmpty(details) && traceEvent.Exception != null)
             {
                 details = Sanitizer.Sanitize(traceEvent.Exception.ToFormattedString());
+                if (
+                    (traceEvent.Exception is FunctionInvocationException ||
+                    traceEvent.Exception is AggregateException ||
+                    traceEvent.Exception is TargetInvocationException) &&
+                    traceEvent.Exception.InnerException != null)
+                {
+                    GetExceptionDetails(traceEvent.Exception.InnerException, out exceptionType, out exceptionMessage, out exceptionStackTrace);
+                }
+                else
+                {
+                    GetExceptionDetails(traceEvent.Exception, out exceptionType, out exceptionMessage, out exceptionStackTrace);
+                }
             }
 
-            _eventGenerator.LogFunctionTraceEvent(traceEvent.Level, subscriptionId, appName, functionName, eventName, source, details, summary);
+            _eventGenerator.LogFunctionTraceEvent(traceEvent.Level, subscriptionId, appName, functionName, eventName, source, details, summary, exceptionType, exceptionMessage, exceptionStackTrace);
+        }
+
+        private void GetExceptionDetails(Exception exception, out string exceptionType, out string exceptionMessage, out string exceptionStackTrace)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException(nameof(exception));
+            }
+            exceptionType = exception.GetType().ToString();
+            exceptionMessage = exception.Message ?? string.Empty;
+            exceptionStackTrace = Sanitizer.Sanitize(exception.ToFormattedString()) ?? string.Empty;
```

Comment:
For the stack trace, is this better than `exception.StackTrace`?

Reply:
Also, we should be sanitizing the message. I don't think the stack trace needs to be sanitized.

Reply:
Ok. Thanks. Will update.